packages/qwik-city/runtime/src/qwik-image-component.tsx

```diff
@@ -0,0 +1,291 @@
+    return undefined;
+  }
+  switch (layout) {
+    // If screen is wider than the max size, image width is the max size,
+    // otherwise it's the width of the screen
+    case `constrained`:
+      return `(min-width: ${width}px) ${width}px, 100vw`;
+
+    // Image is always the same width, whatever the size of the screen
+    case `fixed`:
+      return `${width}px`;
+
+    // Image is always the width of the screen
+    case `fullWidth`:
+      return `100vw`;
+
+    default:
+      return undefined;
+  }
+};
+
+export const getBlurStyles = async ({
+  blur,
+  blurUrlLoader$,
+  src = '',
+}: Pick<QwikImageProps, 'blur' | 'blurUrlLoader$' | 'src'>): Promise<
+  Record<string, string | number>
+> => {
+  if (!blur) {
+    return {};
+  }
+
+  if (!blurUrlLoader$) {
+    throw new Error('Your must specify a blurUrlLoader$ function to use the blur functionality');
+  }
+
+  const [width, height, quality] = [50, 50, 20];
+  return {
+    backgroundSize: 'cover',
+    backgroundPosition: '50% 50%',
+    backgroundRepeat: 'no-repeat',
+    backgroundImage: `url(${await blurUrlLoader$({ src, width, height, quality })})`,
+  };
+};
+
+/**
+ * @alpha
+ */
+export const QwikImage = component$<QwikImageProps>((props) => {
+  const imgRef = useSignal<HTMLImageElement>();
+  const state = useContext(QwikImageContext);
+  const {
+    src = '',
+    blur,
+    blurUrlLoader$,
+    layout,
+    resolutions,
+    ...imageAttributes
+  } = { ...state, ...props };
+  const sizes = getSizes({ ...imageAttributes, layout });
+  const srcset = getSrcSet({ ...imageAttributes, src, layout, resolutions });
+  // In blur mode we use backgroundImage to get the first render
+  const onLoadBlurAttributes = blur ? {} : { src, srcset, sizes };
+
+  const stylesSignal = useSignal({ ...props.style, ...getStyles({ ...imageAttributes, layout }) });
+
+  let width = imageAttributes.width;
+  let height = imageAttributes.height;
+  if (layout === 'fullWidth' || layout === 'constrained') {
+    width = undefined;
+    height = undefined;
+  }
+
+  useTask$(async () => {
+    stylesSignal.value = {
+      ...stylesSignal.value,
+      ...(await getBlurStyles({ blur, blurUrlLoader$, src })),
+    };
+  });
+
+  useBrowserVisibleTask$(() => {
+    if (imgRef.value && blur) {
+      (imgRef.value.decode() || Promise.resolve())
+        .catch(() => {})
+        .then(() => {
+          if (imgRef.value) {
+            /* when the background image is loaded we will
+             * set img html attributes to start src image loading
+             */
+            imgRef.value.setAttribute('src', src);
+            imgRef.value.setAttribute('srcset', srcset);
+            if (sizes) {
+              imgRef.value.setAttribute('sizes', sizes);
+            }
+          }
+        });
+    }
+  });
+
+  return (
+    <img
+      ref={imgRef}
+      decoding="async"
+      {...imageAttributes}
+      {...onLoadBlurAttributes}
+      style={stylesSignal.value}
+      width={width}
```

Comment:
Is this useful? Are these css or image pixels?

Reply:
These are pixels e.g. width=300